tests/spread/build-providers/lxd-basic/task.yaml

```diff
@@ -2,6 +2,7 @@ summary: Build a basic snap using LXD and ensure that it runs
 
 environment:
   SNAP_DIR: ../snaps/make-hello
+  SNAPCRAFT_BUILD_ENVIRONMENT: lxd
```

Comment:
why do we need this? --use-lxd should be enough

Reply:
IIRC (and by that, I mean if the commit message is accurate lol)

SNAPCRAFT_BUILD_ENVIRONMENT: "host" is set by spread.yaml.  Which is now an error due to mismatch.  The option would be to either unset SNAPCRAFT_BUILD_ENVIRONMENT or setting it to "lxd".

Reply:
Can we unset it then? It might bring the expectation that this is the way to do things

Reply:
Sure! Looks like that matches what was done for multipass too.